lib/adapters/parsers/columns.js

```diff
@@ -0,0 +1,129 @@
+// parsing.
+class ColumnParser extends base {
+    // Parse the given line into a set of columns based on whitespace. Returns
+    // an array of objects containing start / end offsets of the column  and the
+    // text value.
+    findColumns(line) {
+        let re = /\S+/gm;
+        let columns = [];
+        let match;
+        while ((match = re.exec(line))) {
+            columns.push({start: match.index, end: re.lastIndex, value: match[0]});
+        }
+        return columns;
+    }
+
+    // Given a set of headers and values parsed into column specs, return an
+    // object containing the best match based on either the start or the end
+    // offsets.
+    getValues(headers, columns, line) {
+        let hidx = 0, cidx = 0;
+
+        var result = {};
+
+        while (hidx < headers.length) {
+            let h = headers[hidx];
+            let c = columns[cidx];
+
+            // Fill in nulls when there are no more columns
+            if (cidx === columns.length) {
+                result[h.value] = null;
+                hidx++;
+
+            // For the last column, grab the full text from the original line to
+            // handle cases like ps output where there can be spaces in the
+            // final column
+            } else if (hidx === headers.length - 1) {
+                result[h.value] = line.substr(c.start);
+                break;
+
+            // If the value text starts anywhere to the left of the column of the
+            // header, then it's a match.
+            } else if (c.start <= h.end) {
+                result[h.value] = c.value;
+                hidx++;
+                cidx++;
+
+            } else if (c.start > h.start) {
+                // in some cases there is no entry for a given column
+                result[h.value] = null;
+                hidx++;
+
+            } else {
+                throw new Error('cannot parse line -- at header ' + hidx + ' ' + 
+                    JSON.stringify(h) + ' column: ' + cidx + ' ' + JSON.stringify(c));
+            }
+        }
+
+        return result;
+    }
+
+    parseStream(stream, emit) {
+        let buffer = [];
+
+        function drain() {
+            let points = buffer;
+            buffer = [];
+            emit(points);
+        }
+
+        return new Promise((resolve, reject) => {
+            let liner = readline.createInterface({
+                input: stream,
+                terminal: false
+            });
+
+            let headers = null;
+
+            liner.on('line', (line) => {
+                this.totalRead++;
+                if (this.doneParsing) {
+                    return;
+                }
+
+                if (!headers) {
+                    headers = this.findColumns(line);
+                    return;
+                }
+
+                let columns = this.findColumns(line);
+                let point = this.getValues(headers, columns, line);
```

Comment:
No, I'm totally wrong, didn't notice we're in an event handler here. So, this is not fine.